File: plugins/oracle/scripts/mcp-fleet/services/atlassian.mjs
// Atlassian site probe.
//
// sooperset/mcp-atlassian consumes per-instance API token + email + site
// URL via CONFLUENCE_URL / CONFLUENCE_USERNAME / CONFLUENCE_API_TOKEN
// (and the JIRA_* trio). One instance per Atlassian site. This sidesteps
// the official remote MCP's known multi-site OAuth collision bug
// (atlassian/atlassian-mcp-server#23, microsoft/vscode#293533).

import { launchProfile } from "../lib/playwright-launcher.mjs";
import { ask, askSecret } from "../lib/prompt.mjs";
import { upsert } from "../lib/workspaces-store.mjs";

export async function probe({ label }) {
  const wsLabel = label || (await ask("Atlassian site label (e.g. 'acme'): "));
  if (!wsLabel) throw new Error("site label is required");

  const siteUrl = await ask("Site URL (e.g. https://acme.atlassian.net): ");
  if (!/^https:\/\/[^/]+\.atlassian\.net\/?$/.test(siteUrl)) {
    throw new Error("Expected an https://<tenant>.atlassian.net URL");
  }

  const email = await ask("Atlassian account email: ");
  if (!email.includes("@")) throw new Error("Expected an email address");

  process.stdout.write(
    `\nLaunching Chrome with an isolated profile for atlassian/${wsLabel}.\n` +
    `Sign in, then create an API token at the page that opens.\n\n`,
  );

  const { close } = await launchProfile({
    service: "atlassian",
    workspace: wsLabel,
    startUrl: "https://id.atlassian.com/manage-profile/security/api-tokens",
  });

  const token = await askSecret("Paste the API token (input hidden): ");
  await close();

  if (!token) throw new Error("API token is required");

  const baseUrl = siteUrl.replace(/\/$/, "");
  return upsert({
    service: "atlassian",
    label: wsLabel,
    kind: "atlassian-api-token",
    credentials: {
      CONFLUENCE_URL: `${baseUrl}/wiki`,
      CONFLUENCE_USERNAME: email,
      CONFLUENCE_API_TOKEN: token,
      JIRA_URL: baseUrl,
      JIRA_USERNAME: email,
      JIRA_API_TOKEN: token,
    },
    profileDir: "",
  });
}

if (import.meta.url === `file://${process.argv[1]}`) {
  probe({ label: process.argv[2] }).then(
    (e) => { process.stdout.write(`Persisted atlassian site ${e.label}\n`); process.exit(0); },
    (e) => { process.stderr.write(`${e.message}\n`); process.exit(1); },
  );
}

File: plugins/oracle/scripts/mcp-fleet-py/mcp_fleet/services/atlassian.py
"""Atlassian site probe.

sooperset/mcp-atlassian consumes per-site API token + email + site URL.
One workspace = one Atlassian site (covering both Jira and Confluence on
that site -- the same API token authenticates against both products).

After paste we GET /rest/api/3/myself with basic auth (email:token) to
validate. We then probe /wiki/rest/api/user/current to confirm Confluence
access on the same site; failure there is a WARN not an ERROR (a site
might have only one product enabled).
"""

from __future__ import annotations

import base64
import re

import httpx

from .. import store
from ..paths import profile_dir
from .base import ProbeContext, ProbeError


URL_RE = re.compile(r"^https://[^/]+\.atlassian\.net/?$")


async def _validate(site_url: str, email: str, token: str, ctx: ProbeContext) -> tuple[str, bool, bool]:
    """Returns (display_name, jira_ok, confluence_ok)."""
    base = site_url.rstrip("/")
    auth = base64.b64encode(f"{email}:{token}".encode()).decode()
    headers = {"Authorization": f"Basic {auth}", "Accept": "application/json"}

    async with httpx.AsyncClient(timeout=15, headers=headers) as client:
        jr = await client.get(f"{base}/rest/api/3/myself")
        if jr.status_code != 200:
            raise ProbeError(f"Atlassian Jira rejected the token ({jr.status_code}): {jr.text[:200]}")
        display = (jr.json() or {}).get("displayName") or email

        jira_ok = True
        confluence_ok = False
        try:
            cr = await client.get(f"{base}/wiki/rest/api/user/current")
            confluence_ok = cr.status_code == 200
        except Exception:
            confluence_ok = False

    return display, jira_ok, confluence_ok


async def probe(label: str, ctx: ProbeContext) -> store.Workspace:
    if not label:
        raise ProbeError("site label is required")

    site_url = (await ctx.paste_text(
        "Site URL (e.g. https://acme.atlassian.net)"
    )).strip()
    if not URL_RE.match(site_url):
        raise ProbeError("expected https://<tenant>.atlassian.net")

    email = (await ctx.paste_text("Atlassian account email")).strip()
    if "@" not in email:
        raise ProbeError("expected an email address")

    ctx.log(f"opening Chromium at id.atlassian.com/manage-profile/security/api-tokens for atlassian/{label}")
    from .browser_helper import open_url_in_profile
    async with open_url_in_profile(
        service="atlassian",
        workspace=label,
        url="https://id.atlassian.com/manage-profile/security/api-tokens",
    ):
        token = (await ctx.paste_secret("Paste the API token")).strip()

    if not token:
        raise ProbeError("API token is required")

    ctx.log("validating against Jira + Confluence APIs ...")
    display, jira_ok, confluence_ok = await _validate(site_url, email, token, ctx)
    products = []
    if jira_ok:
        products.append("Jira")
    if confluence_ok:
        products.append("Confluence")
    else:
        ctx.log("NOTE: Confluence is not reachable on this site/token; Jira-only binding.")
    ctx.log(f"validated -- {display} ({' + '.join(products) or 'unknown product'})")

    base = site_url.rstrip("/")
    return store.upsert(store.Workspace(
        service="atlassian",
        label=label,
        kind="atlassian-api-token",
        credentials={
            "CONFLUENCE_URL": f"{base}/wiki",
            "CONFLUENCE_USERNAME": email,
            "CONFLUENCE_API_TOKEN": token,
            "JIRA_URL": base,
            "JIRA_USERNAME": email,
            "JIRA_API_TOKEN": token,
        },
        profileDir=str(profile_dir("atlassian", label)),
        discoveredName=f"{display} @ {base}",
    ))
